### PR TITLE
Latency improvement

### PR DESCRIPTION
**Pull request by Labonneguigue**
PR based upon #8 and #10.

Exactly like this guy suggests, I have issues when I enable the camera.
https://carnd.slack.com/archives/C6NVDVAQ3/p1520590992000332 

The green waypoints on the road start lagging more and more behind the car.

Do you have such behavior ? 

This commit reduces the frequency of publishers. The following diagrams are obtained with the last line in styx.launch and running `rqt_graph`.

First with camera enabled and then without camera enabled.

rosgraph_camera_optimisation
rosgraph_no_camera_optimisation

### CODE REVIEW

**Comment by manu291** [comment]
I see that the traffic light detector launcher is still commented out `styx.launch`, but you mentioned in PR #8 that the changes introduced in the present PR solves the issue. Is that correct? If it does then why not uncommenting the launchers in `styx.launch`?

File: ros/src/waypoint_loader/waypoint_loader.py [hunk 1]
```diff
@@ -72,11 +72,14 @@ def decelerate(self, waypoints):
         return waypoints
 
     def publish(self, waypoints):
-        lane = Lane()
-        lane.header.frame_id = '/world'
-        lane.header.stamp = rospy.Time(0)
-        lane.waypoints = waypoints
-        self.pub.publish(lane)
+        rate = rospy.Rate(1)
+        while not rospy.is_shutdown():
```
Comment (manu291) [inline ros/src/waypoint_loader/waypoint_loader.py]: Why do you publish the way points in that while loop?
They are needed only once by the other modules.
The changes here make `tl_detector.py` load the way points and find the stop line endlessly.
As a result everything else is stuck, the car is not able drive and no prediction is made.
On the other hand if I revert to the previous version of that file (`waypoint_loader.py`), everything works fine (car driving and traffic light predictions).

**Comment by manu291** [comment]
Oddly enough the changes in `waypoint_loader.py` do no cause the problem mentioned above when using the branch of PR #12.

File: ros/src/waypoint_loader/waypoint_loader.py (hunk 1 above)
Comment (manu291) [reply ros/src/waypoint_loader/waypoint_loader.py]: Oddly enough the changes in waypoint_loader.py do no cause the problem mentioned above when using the branch of PR #12.

**Review by manu291 — APPROVED** [review]

File: ros/src/waypoint_loader/waypoint_loader.py (hunk 1 above)
Comment (Labonneguigue) [reply ros/src/waypoint_loader/waypoint_loader.py]: Good catch ! I didn't mean to do that. Correcting right now.

**Comment by Labonneguigue** [comment]
Having corrected the error you mentioned, I merge this PR.

**Pull request merged**